src/org/labkey/test/pages/experiment/CreateSampleSetPage.java

```diff
@@ -0,0 +1,113 @@
+    }
+
+    public static CreateSampleSetPage beginAt(WebDriverWrapper driver, String containerPath)
+    {
+        driver.beginAt(WebTestHelper.buildURL("experiment", containerPath, "createSampleSet"));
+        return new CreateSampleSetPage(driver.getDriver());
+    }
+
+    public CreateSampleSetPage setName(String name)
+    {
+        elementCache().nameInput.set(name);
+        return this;
+    }
+
+    public String getName()
+    {
+        return elementCache().nameInput.get();
+    }
+
+    public CreateSampleSetPage setDescription(String desc)
+    {
+        elementCache().descriptionInput.set(desc);
+        return this;
+    }
+
+    public String getDescrption()
+    {
+        return elementCache().descriptionInput.get();
+    }
+
+    public CreateSampleSetPage setNameExpression(String nameExp)
+    {
+        elementCache().nameExpressionInput.set(nameExp);
+        return this;
+    }
+
+    public String getNameExpression()
+    {
+        return elementCache().nameExpressionInput.get();
+    }
+
+    public CreateSampleSetPage addParentColumnAlias(String importHeader, String materialInputName)
+    {
+        int countOfInputs = Locator.tagWithName("input", "importAliasKeys").findElements(getDriver()).size();
+
+        click(Locator.linkWithText("add parent column import alias"));
+        waitFor(()-> Locator.tagWithName("input", "importAliasKeys").findElements(getDriver()).size() > countOfInputs, 1000);
+
+        List<WebElement> importAliasInputs = Locator.tagWithName("input", "importAliasKeys").findElements(getDriver());
+        List<WebElement> importAliasSelects = Locator.tagWithName("select", "importAliasValues").findElements(getDriver());
+
+        int index = importAliasInputs.size() - 1;
+        WebElement aliasInput = importAliasInputs.get(index);
+        WebElement aliasSelect = importAliasSelects.get(index);
+
+        setFormElement(aliasInput, importHeader);
+        selectOptionByTextContaining(aliasSelect, materialInputName);
+
+        return this;
+    }
+
+    public EditDomainPage clickCreate()
+    {
+        clickAndWait(elementCache().createButton);
+        return new EditDomainPage(getDriver());
+    }
+
+    public void clickCancel()
+    {
+        clickAndWait(elementCache().cancelButton);
+    }
+
+    protected ElementCache newElementCache()
+    {
+        return new ElementCache();
+    }
+
+    protected class ElementCache extends LabKeyPage.ElementCache
+    {
+        Input nameInput = Input.Input(Locator.name("name"), getDriver())
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
+        Input nameExpressionInput = Input.Input(Locator.name("nameExpression"), getDriver())
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
+        Input descriptionInput = Input.Input(Locator.name("description"), getDriver())
+                .timeout(WAIT_FOR_JAVASCRIPT).findWhenNeeded(this);
+
```

Comment:
Should there be an element, or a locator, for the <div class="labkey-error"> tag? And then by extension a method that gets the error message, maybe after clicking "Create".

Reply:
This page is largely out of scope. I was just cleaning up existing functionality.